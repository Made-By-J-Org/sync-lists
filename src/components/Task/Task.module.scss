@import '../../styles/colors';

.taskWrapper {
  height: 40px;
  padding-top: 10px;

  > :last-child {
    margin-left: 10px;
  }
}

.checkbox {
  display: inline-block;
  position: relative;
  top: 4px;
  width: 14px;
  height: 14px;
  border: 1px solid $checkbox-border-color;
  border-radius: 3px;
  cursor: pointer;
}

.checkboxSelected {
  background: $checkbox-checked-color;
}

.checkboxSelected:after {
  content: '\2713';
  color: white;
  position: relative;
  top: -3px;
}


