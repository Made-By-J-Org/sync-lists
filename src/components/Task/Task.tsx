import React, {useState} from "react";

import {TaskSchema} from "../../api/api.mock";

import styles from "./Task.module.scss";

interface TaskProps {
  task: TaskSchema;
}

const Task = ({task}: TaskProps): React.ReactElement => {
  const [selected, setSelected] = useState(false);
  return (
    <div
      data-testid='task-wrapper'
      className={styles.taskWrapper}
      onClick={() => setSelected(prev => !prev)}
    >
      <span
        data-testid='task-wrapper-checkbox'
        className={`
          ${selected ? styles.checkboxSelected : ''}
          ${styles.checkbox}
        `}
      />
      <span>
        {task.description}
      </span>
    </div>
  )
}

export default Task;
