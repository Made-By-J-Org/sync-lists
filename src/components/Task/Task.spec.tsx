import React from "react";
import {render} from "@testing-library/react";
import Task from "./Task";
import userEvent from '@testing-library/user-event'

describe("Task component", () => {
  const task = {
    checked: false,
    description: "Sample task",
    value: 10
  };

  it('render component', () => {
    const {baseElement} = render(<Task task={task}/>);
    expect(baseElement).toBeInTheDocument();
  });

  it('match snapshot', () => {
    const {asFragment} = render(<Task task={task}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the task description", () => {
    const {getByText} = render(<Task task={task}/>);
    expect(getByText("Sample task")).toBeInTheDocument();
  });

  it("should toggle the selected state on click", async () => {
    const user = userEvent.setup()
    const {getByTestId} = render(<Task task={task}/>);
    const taskWrapperCheckbox = getByTestId("task-wrapper-checkbox");

    expect(taskWrapperCheckbox.getAttribute('class')).not.toMatch(/checkboxSelected/gi)
    await user.click(taskWrapperCheckbox);
    expect(taskWrapperCheckbox.getAttribute('class')).toMatch(/checkboxSelected/gi)
    await user.click(taskWrapperCheckbox);
    expect(taskWrapperCheckbox.getAttribute('class')).not.toMatch(/checkboxSelected/gi)
  });
});
