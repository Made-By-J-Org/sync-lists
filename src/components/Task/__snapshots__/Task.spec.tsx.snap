// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task component > match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_taskWrapper_416c64"
    data-testid="task-wrapper"
  >
    <span
      class="
          
          _checkbox_416c64
        "
      data-testid="task-wrapper-checkbox"
    />
    <span>
      Sample task
    </span>
  </div>
</DocumentFragment>
`;
